elements/campaign-display/fields/campaign-request-field.js

```diff
@@ -19,7 +19,7 @@ const CampaignRequestField = {
 
     fetchCampaignSuccess(state, response, store) {
       state.requestInFlight = false;
-      store.actions.handleFetchComplete(response);
+      setImmediate(() => store.actions.handleFetchComplete(response));
```

Comment:
So it wouldn't work as-is, but if we disallow nested actions within the same field, we can have a simple change that allows this it happen immediately.